prime-router/docs/design/proposals/0025-handling-missing-data.md

```diff
@@ -0,0 +1,36 @@
+# Proposal for handling missing timezones in messages
+
+## Background
+
+ReportStream is currently facing an ongoing issue regarding missing timezones in date times sent by some senders. 
+Certain STLTs require the conversion of all datetime values to their respective local timezones. In cases where 
+timezones are absent, ReportStream defaults the datetime to UTC and then converts it to the receiver's local timezone.
+This can be an issue since ReportStream is essentially making up data and could be reporting incorrect data.
+To tackle this issue, there are alternative approaches to handling missing timezones instead of resorting to UTC 
+as the default. We also propose implementing a logging mechanism to notify the STLT when such discrepancies occur.
+
+## Possible Implementations
+
+We should encourage Senders to include timezones for date times whenever they're missing from messages and issue 
+a warning message whenever this information is absent, ensuring they are aware of the issue.
+
+In cases where a sender cannot provide a timezone for a datetime, ReportStream will need to determine the most appropriate timezone for the Sender. This can be achieved through the following methods:
+
+* Request the sender to specify a default timezone for their date times, which could be added as a sender transform.
+* Retrieve the timezone by looking at the Patient's address. We would need to add logic for this.
+* Check if timezone information is available in other date times in the message.
```

Comment:
I think this should be handled at onboarding if possible. If the sending application can generate a date/time stamp with appropriate timezone offset for one field it should be able to do so for others.

Reply:
Agreed with @GilmoreA6

Reply:
I think this was the case for some of the ELIMS messages. They couldn't provide a timezone for one of the fields even though they were generating for others and had some reasons why they couldn't do that.
Check row 24 in the ELIMS gap analysis here https://cdc.sharepoint.com/:x:/r/teams/ReportStream/_layouts/15/Doc.aspx?sourcedoc=%7BFAED6C19-BA41-4F03-A148-6C8490228A1C%7D&file=HL7%20IN%20vs%20HL7%20OUT%20_%20Gap_Analysis.xlsx&action=default&mobileredirect=true

Reply:
We could still handle this with a transform at onboarding so I think that's fine.